devicetypes/smartthings/zwave-lock.src/zwave-lock.groovy

```diff
@@ -1365,6 +1866,579 @@ def deleteCode(codeID) {
+	log.trace "[DTH] Executing 'clearOldSchedule()' by ${device.displayName} with codeID := $codeID"
+	def cmds = []
+	
+	if (codeID instanceof String) {
+		codeID = Integer.parseInt(codeID, 10)
+	}
+	
+	if (isKwiksetLock()) {
+		cmds << secure(zwave.configurationV2.configurationSet(parameterNumber: codeID, configurationValue: [1]))
+	} else {
+		if (device.currentValue("numberOfSlotsDailyRepeating")) {
+			cmds << secure(zwave.scheduleEntryLockV3.scheduleEntryLockDailyRepeatingSet(scheduleSlotId: 1, setAction: 0, userIdentifier: codeID))
+		}
+		if (device.currentValue("numberOfSlotsWeekDay")) {
+			for (def i = 0; i <= 6; i++) {
+				cmds << secure(zwave.scheduleEntryLockV3.scheduleEntryLockWeekDaySet(setAction: 0, scheduleSlotId: (i + 1), userIdentifier: codeID))
+			}
+		}
+		if (device.currentValue("numberOfSlotsYearDay")) {
+			cmds << secure(zwave.scheduleEntryLockV3.scheduleEntryLockYearDaySet(scheduleSlotId: 1, setAction: 0, userIdentifier: codeID))
+		}
+	}
+	
+	if (cmds) {
+		cmds = cmds.flatten()
+		log.trace "[DTH] 'clearOldSchedule()' returning with cmds := $cmds"
+		return cmds
+	}
+	return null
+}
+
+/**
+ * API end-point for changing the user type of a code slot to type owner. This is applicable only for Kwikset locks.
+ *
+ * @param codeID: The code slot number
+ *
+ * @returns cmds: The commands fired for changing the user type
+ */
+def changeUserTypeToOwner(codeID) {
+	if (isKwiksetLock()) {
+		if (codeID != null) {
+			def cmds = []
+			cmds << secure(zwave.configurationV2.configurationSet(parameterNumber: codeID, configurationValue: [1]))
+			cmds << secure(zwave.configurationV2.configurationGet(parameterNumber: codeID))
+			cmds = delayBetween(cmds, 2000)
+			return cmds
+		}
+		return null
+	}
+	return null
+}
+
+/**
+ * Returns the command(s) for setting daily repeating schedule
+ *
+ * @return The command(s) for setting daily repeating schedule
+ */
+private def getDailyRepeatingSetScheduleCmd(schSettings) {
+	if (device.currentValue("numberOfSlotsDailyRepeating")) {
+		log.trace "[DTH] Executing 'getDailyRepeatingSetScheduleCmd()' by ${device.displayName} with schSettings := $schSettings"
+		def cmds = []
+		
+		def codeID = Integer.parseInt(schSettings.codeId, 10)
+		
+		def startTime = getWeekDayHoursAndMinutes(schSettings.startTime)
+		def startHour = startTime.hour
+		def startMin = startTime.minute
+		
+		def endTime = getWeekDayHoursAndMinutes(schSettings.endTime)
+		def endHour = endTime.hour
+		def endMin = endTime.minute
+		
+		def durMap = getDurationInHoursAndMinutes(startHour, startMin, endHour, endMin)
+		def durMin = durMap.durMin
+		def durHour = durMap.durHour
+		
+		// For weekDayBitMask, the order of days is Sat, Fri, Thu, Wed, Tue, Mon, Sun with Sat as the most significant bit
+		short weekDayBitMask = schSettings.weekDayBitMask
+		
+		state["setDailyRepeatingStartTime$codeID"] = schSettings.startTime
+		state["setDailyRepeatingEndTime$codeID"] = schSettings.endTime
```

Comment:
This is the only time you're using `endTime` in this scope. It doesn't need to be split into constituent fields.